openff/interchange/_tests/unit_tests/interop/gromacs/export/test_export.py

```diff
@@ -414,6 +418,84 @@ def test_exisiting_mol0_names_overwritten(self, name, sage, ethanol, cyclohexane
 
         assert [*parmed.load_file("tmp.top").molecules.keys()] == ["MOL0", "MOL1"]
 
+    @pytest.mark.parametrize("name", ["MOL0", "MOL222", ""])
+    def test_roundtrip_with_combine(
+        self,
+        name,
+        sage_unconstrained,
+        ethanol,
+        cyclohexane,
+        monkeypatch,
+    ):
+        monkeypatch.setenv("INTERCHANGE_EXPERIMENTAL", "1")
+
+        for molecule in [ethanol, cyclohexane]:
+            molecule.name = name
+            molecule.generate_conformers(n_conformers=1)
+
+        # translate one by 1 nm
+        cyclohexane._conformers = [
+            cyclohexane.conformers[0] + Quantity([0, 0, 1], "nanometer"),
+        ]
+
+        ethanol_interchange = sage_unconstrained.create_interchange(
+            ethanol.to_topology(),
+        )
+        ethanol_interchange.box = [4, 4, 4]
+
+        cyclohexane_interchange = sage_unconstrained.create_interchange(
+            cyclohexane.to_topology(),
+        )
+        cyclohexane_interchange.box = [4, 4, 4]
+
+        combined = ethanol_interchange.combine(cyclohexane_interchange)
+
+        ethanol_interchange.to_gromacs("ethanol", decimal=8)
+        cyclohexane_interchange.to_gromacs("cyclohexane", decimal=8)
+        combined.to_gromacs("combined", decimal=8)
+
+        converted_ethanol = Interchange.from_gromacs(
+            "ethanol.top",
+            "ethanol.gro",
+        )
+        converted_cyclohexane = Interchange.from_gromacs(
+            "cyclohexane.top",
+            "cyclohexane.gro",
+        )
+
+        converted_combined = Interchange.from_gromacs(
+            "combined.top",
+            "combined.gro",
+        )
+
+        combined_from_converted = converted_ethanol.combine(converted_cyclohexane)
+        combined_from_converted["vdW"].cutoff = combined["vdW"].cutoff
+        converted_combined["vdW"].cutoff = combined["vdW"].cutoff
+
+        assert numpy.allclose(converted_combined.positions, combined.positions)
+        assert numpy.allclose(
+            converted_combined.positions,
+            combined_from_converted.positions,
+        )
+        assert numpy.allclose(converted_combined.box, combined.box)
+        assert numpy.allclose(converted_combined.box, combined_from_converted.box)
+
+        get_gromacs_energies(combined).compare(
+            get_gromacs_energies(converted_combined),
+            tolerances={
+                "Electrostatics": 0.05 * unit.kilojoule / unit.mol,
+            },
+        )
+        try:
+            get_gromacs_energies(combined).compare(
+                get_gromacs_energies(combined_from_converted),
+                tolerances={
+                    "Electrostatics": 0.05 * unit.kilojoule / unit.mol,
+                },
+            )
```

Comment:
This is failing pretty catastrophically, not yet sure why

Reply:
Also this is failing

```python
get_openmm_energies(converted_ethanol.combine(converted_cyclohexane)).compare(get_openmm_energies(converted_cyclohexane.combine(converted_ethanol)), tolerances={"Nonbonded": Quantity(0.01, "kilojoule/mole")})
```

Reply:
It's probably H-X bonds that aren't being parsed